Applications/Utils/FileConverter/TecPlotTools.cpp

```diff
@@ -0,0 +1,410 @@
+        std::string const extension(file_name.substr(delim_pos, std::string::npos));
+
+        val_count = 0;
+        val_total = 0;
+        INFO("Writing section #%i", write_count);
+        out.close();
+        out = std::ofstream(base_name + std::to_string(write_count++) + extension);
+    }
+}
+
+/// Writes one section/zone into a seperate OGS-mesh
+int writeDataToMesh(std::string const& file_name,
+                    std::size_t& write_count,
+                    std::vector<std::string> const& vec_names,
+                    std::vector< std::vector<double> > const& scalars,
+                    std::pair<std::size_t, std::size_t> const& dims)
+{
+    double cellsize = 0;
+    for (std::size_t i=0; i<vec_names.size(); ++i)
+    {
+        if (vec_names[i] == "x" || vec_names[i] == "X")
+        {
+            cellsize = scalars[i][1] - scalars[i][0];
+            break;
+        }
+    }
+
+    if (cellsize == 0)
+    {
+        ERR ("Cell size not found. Aborting...");
+        return -4;
+    }
+
+    GeoLib::Point origin (0, 0, 0);
+    GeoLib::RasterHeader header { dims.first, dims.second, 1, origin, cellsize, -9999 };
+
+    std::unique_ptr<MeshLib::Mesh> mesh (MeshLib::RasterToMesh::convert(
+        scalars[0].data(),
+        header,
+        MeshLib::MeshElemType::QUAD,
+        MeshLib::UseIntensityAs::DATAVECTOR,
+        vec_names[0]));
+    MeshLib::Properties& properties = mesh->getProperties();
+    for (std::size_t i=1; i<vec_names.size(); ++i)
+    {
+        auto* const prop = properties.createNewPropertyVector<double>(
+            vec_names[i], MeshLib::MeshItemType::Cell, 1);
+        if (!prop)
+        {
+            ERR("Error creating array \"%s\".", vec_names[i].c_str());
+            return -5;
+        }
+        prop->reserve(scalars[i].size());
+        std::copy(scalars[i].cbegin(), scalars[i].cend(), std::back_inserter(*prop));
+    }
+
+    std::size_t const delim_pos(file_name.find_last_of("."));
+    std::string const base_name(file_name.substr(0, delim_pos + 1));
+    std::string const extension(file_name.substr(delim_pos, std::string::npos));
+
+    INFO("Writing section #%i", write_count);
+    MeshLib::IO::VtuInterface vtu(mesh.get());
+    vtu.writeToFile(base_name + std::to_string(write_count++) + extension);
+    return 0;
+}
+
+/// If a geometry-section is encountered, it is currently ignored
+void skipGeometrySection(std::ifstream& in, std::string& line)
+{
+    while (getline(in, line))
+    {
+        if ((line.find("TITLE") != std::string::npos) ||
+            (line.find("VARIABLES") != std::string::npos) ||
+            (line.find("ZONE") != std::string::npos))
+            return;
+    }
+}
+
+/// Splits a TecPlot file containing multiple sections/zones into seperate files
+int splitFile(std::ifstream& in, std::string file_name)
+{
+    std::ofstream out;
+    std::string line("");
+    std::string name;
+    std::pair<std::size_t, std::size_t> dims(0,0);
```

Comment:
`dims` could be more local, it is used as a temporary variabele while parsing the zone.